packages/server-core/src/projects/project/project-helper.ts

```diff
@@ -1634,14 +1637,20 @@ export const getCommitSHADate = async (projectName: string): Promise<{ commitSHA
   }
 }
 
-export const deleteProjectFilesInStorageProvider = async (projectName: string, storageProviderName?: string) => {
+export const deleteProjectFilesInStorageProvider = async (
+  app: Application,
+  projectName: string,
+  storageProviderName?: string
+) => {
   const storageProvider = getStorageProvider(storageProviderName)
   try {
     const existingFiles = await getFileKeysRecursive(`projects/${projectName}`)
     if (existingFiles.length) {
       await Promise.all([
```

Comment:
I think this should not be parallel rather first it should async delete resource and then add the entry to invalidation service.

From:
```
await Promise.all([
  storageProvider.deleteResources(existingFiles),
  app.service(invalidationPath).create({
    path: `projects/${projectName}*`
  })
])
```

To:
```
await storageProvider.deleteResources(existingFiles),
await app.service(invalidationPath).create({
 path: `projects/${projectName}*`
})

```